File: pg-client/src/pg/client/sql.clj
(ns pg.client.sql)


(defn isolation-level->sql [level]
  (case level

    (:SERIALIZABLE
     :serializable
     "SERIALIZABLE"
     "serializable"
     SERIALIZABLE
     serializable)
    "SERIALIZABLE"

    (:REPEATABLE-READ
     :repeatable-read
     "REPEATABLE-READ"
     "repeatable-read"
     REPEATABLE-READ
     repeatable-read)
    "REPEATABLE READ"

    (:READ-COMMITTED
     :read-committed
     "READ-COMMITTED"
     "read-committed"
     READ-COMMITTED
     read-committed)
    "READ COMMITTED"

    (:READ-UNCOMMITTED
     :read-uncommitted
     "READ-UNCOMMITTED"
     "read-uncommitted"
     READ-UNCOMMITTED
     read-uncommitted)
    "READ UNCOMMITTED"

    (throw
     (ex-info (format "Wrong isolation level: %s" level)
              {:level level}))))


(defn set-tx [{:keys [isolation-level
                      read-only?]}]

  (when (or isolation-level read-only?)
    (with-out-str
      (print "SET TRANSACTION")
      (when isolation-level
        (print " ISOLATION LEVEL ")
        (print (isolation-level->sql isolation-level)))
      (when read-only?
        (print " READ ONLY")))))

File: pg-client2/src/com/github/igrishaev/util/SQL.java
package com.github.igrishaev.util;

import com.github.igrishaev.enums.TxLevel;

public class SQL {

    public static String quoteChannel (final String sql) {
        return String.format("\"%s\"", sql.replaceAll("\"", "\"\""));
    }

    public static String SQLSetTxReadOnly = "SET TRANSACTION READ ONLY";

    public static String SQLSetTxLevel (final TxLevel level) {
        return String.format("SET TRANSACTION ISOLATION LEVEL %s", level.getCode());
    }

    public static void main (final String[] args) {
        System.out.println(quoteChannel("aa\"a'aa"));
    }

}
